### PR TITLE
network requests are now shared

### PR DESCRIPTION
**Pull request by esotrope**
before

image

after

image

### CODE REVIEW

File: webapp/src/main/webapp/src/app/admin/tenant/component/property-override-table/property-override-table.component.html [hunk 1]
```diff
@@ -77,7 +77,9 @@
                 class="fa fa-undo fa-md"
                 *ngIf="overrides.value !== overrides.originalValue && !readonly"
                 (click)="resetClicked(overrides)"
-                popover="{{ 'tenant-config.reset-info' | translate }}"
+                popover="{{
+                  'tenant-configuration-override-form.popover.reset' | translate
```
Comment (esotrope) [inline webapp/src/main/webapp/src/app/admin/tenant/component/property-override-table/property-override-table.component.html]: misc message cleanup that i missed

File: webapp/src/main/webapp/src/app/admin/tenant/pages/tenant/tenant.component.html [hunk 2]
```diff
@@ -1,22 +1,32 @@
...
-    "
-    (create)="onCreate($event)"
-    (update)="onUpdate($event)"
-    (delete)="onDelete($event)"
-  ></app-tenant-form>
-</ng-container>
-<ng-template #loading>
-  <spinner></spinner>
-</ng-template>
+<!-- TODO investigate why this spinner display logic isn't working -->
```
Comment (esotrope) [inline webapp/src/main/webapp/src/app/admin/tenant/pages/tenant/tenant.component.html]: the loading spinner logic did not work in conjunction with the `share()` operator - have no clue why yet...

**Comment by ftdholbrook** [comment]
This is cool I noticed it was spamming some of the same calls

File: webapp/src/main/webapp/src/app/admin/tenant/service/tenant.service.ts [hunk 3]
```diff
@@ -51,15 +55,15 @@ export class TenantService {
...
       .get(DataSetsRoute)
       .pipe(catchError(ResponseUtils.throwError));
   }
 
   getAll(type: TenantType): Observable<TenantConfiguration[]> {
     const tenants$ = this.dataService.get(ResourceRoute, { params: { type } });
-    const defaults$ = this.dataService.get(DefaultsRoute);
-    const dataSets$ = this.dataService.get(DataSetsRoute);
+    const defaults$ = this.cachingDataService.get(DefaultsRoute);
+    const dataSets$ = this.cachingDataService.get(DataSetsRoute);
```
Comment (esotrope) [inline webapp/src/main/webapp/src/app/admin/tenant/service/tenant.service.ts]: @ftdholbrook can system defaults and data sets be cached until a browser tab is refreshed? i assume so but let me know if that is wrong

Comment (ftdholbrook) [reply webapp/src/main/webapp/src/app/admin/tenant/service/tenant.service.ts]: defaults changing would probably require restarting the web server.  Datasets same thing, any way to tie the cache to the session?

Comment (esotrope) [reply webapp/src/main/webapp/src/app/admin/tenant/service/tenant.service.ts]: yeah we technically can however we don't do client-side session caching for users atm - if we do session level caching we do it on the backend atm

**Pull request merged**